Adding a `used-types` rule (#1058)
gajus/eslint-plugin-jsdoc: .README/rules/mark-used.md

> # `mark-used`

Author: I would some feedback on the name. It doesn't convey that this has to do with "types" being marked as used. An alternative would be `used-types` as a tribute to the `valid-types` rule, which already exists in the package.

Reviewer: `used-types` sounds good to me

Author: Pushed a rename to "used-types".

{"gitdown": "contents", "rootId": "mark-used"}

> Marks all types referenced in `{@link}` tags as used.

Author: ```suggestion
Marks all types referenced from jsdoc tags as used.
```

Reviewer: One nit: change "jsdoc" to "JSDoc"
